compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmCodeGenUtil.java

```diff
@@ -202,9 +202,15 @@ public static String rewriteVirtualCallTypeName(String value, BType objectType)
         objectType = getImpliedType(objectType);
         // The call name will be in the format of`objectTypeName.funcName` for attached functions of imported modules.
         // Therefore, We need to remove the type name.
+        Name originalName = objectType.tsymbol.originalName;
         if (!objectType.tsymbol.name.value.isEmpty() && value.startsWith(objectType.tsymbol.name.value)) {
             value = value.replace(objectType.tsymbol.name.value + ".", "").trim();
         }
+        // The call name will be in the format of`objectTypeOriginalName.funcName` for attached functions of
+        // object definitions. Therefore, We need to remove it.
+        if (originalName != null && !originalName.value.isEmpty() && value.startsWith(originalName.value)) {
```

Comment:
Will there be a case where reaching both the if conditions is important or can we change the second if condition to an else condition?